Reviewer/src/main/java/com/softserveinc/reviewer/health/TimeLimitedHealthCheck.java

```diff
@@ -0,0 +1,49 @@
+package com.softserveinc.reviewer.health;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.softserveinc.reviewer.config.MongoConfiguration;
+
+public class TimeLimitedHealthCheck extends HealthCheck {
+    private final long timeout;
+    private final TimeUnit timeUnit;
+    private final MongoConfiguration mongoConfiguration;
+    private final String[] urls;
+    private final ExecutorService executorService = Executors.newFixedThreadPool(4);
+
+    public TimeLimitedHealthCheck(long timeout, TimeUnit timeUnit, MongoConfiguration mongoConfiguration, String... urls) {
+        this.timeout = timeout;
+        this.timeUnit = timeUnit;
+        this.mongoConfiguration = mongoConfiguration;
+        this.urls = urls;
+    }
+
+    @Override
+    protected Result check() throws Exception {
+        Set<Callable<Boolean>> callables = new HashSet<>();
+        callables.add(new MongoDBHealthCheck(mongoConfiguration));
+        Arrays.stream(urls).forEach(x -> callables.add(new SimpleURLHealthCheck(x)));
+        try {
+            List<Future<Boolean>> futures = executorService.invokeAll(callables, timeout, timeUnit);
+            for (Future<Boolean> future : futures) {
+                if (future.isCancelled() || !future.get()) {
+                    return Result.unhealthy("At least one of the Services are unhealthy");
+                }
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            return Result.unhealthy("Exception occurred during checking Services");
```

Comment:
I believe that we can let dropwizard handle those exceptions. No need to do it manually,